src/block.rs

```diff
@@ -128,6 +138,24 @@ impl<T: Serialize + Clone> Block<T> {
     pub fn payload(&self) -> &T {
         &self.payload
     }
+
+    /// Return the block state given a set of valid voters.
+    pub fn get_block_state(
+        &self,
+        valid_voters: &BTreeSet<PeerId>,
+    ) -> Result<BlockState, RoutingError> {
+        let total_age = valid_voters
+            .iter()
+            .map(|peer_id| peer_id.age() as usize)
+            .sum();
+        if self.proofs.len() == valid_voters.len() {
```

Comment:
You don't need total age calculation for this branch. Just put it on the top and return. Then get the total age followed by the rest of the logic.